syntax = "proto2";

service ArmazenamentoChaveValorDistribuido{
    rpc Inserir(Chave_Valor) returns (FlagRetorno);
    
    rpc Consultar(Chave) returns (Valor);

    rpc Ativar(StringIDServico) returns (FlagRetorno);

    rpc Terminar(ParamVazio) returns (FlagRetorno);
}

message FlagRetorno{
    required int32 flag = 1;
}

message Chave_Valor{
    required int32 chave = 1;
    required string valor = 2;
}

message Chave{
    required int32 chave = 1;
}

message Valor{
    required string valor = 1;
}

message StringIDServico{
    required string idServico = 1;
}

message ParamVazio{}